index/generator/library/library.go

```diff
@@ -151,7 +164,11 @@ func parseDevfileRegistry(registryDirPath string, force bool) ([]schema.Schema,
 			// Index component validation
 			err := validateIndexComponent(indexComponent, schema.StackDevfileType)
 			if err != nil {
-				return nil, fmt.Errorf("%s index component is not valid: %v", devfileDir.Name(), err)
+				if _, ok := err.(*MissingArchError); !ok {
+					return nil, fmt.Errorf("%s index component is not valid: %v", devfileDir.Name(), err)
+				}
+				// log to the console as FYI if the devfile has no architectures
+				fmt.Printf("the %s devfile has no architecture(s) mentioned\n", indexComponent.Name)
```

Comment:
Can't we just print out `err.Error()` here?

Reply:
good point, done